sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RewriteSubquerySuite.scala

```diff
@@ -33,23 +34,44 @@ class RewriteSubquerySuite extends PlanTest {
       Batch("Rewrite Subquery", FixedPoint(1),
         RewritePredicateSubquery,
         ColumnPruning,
+        InferFiltersFromConstraints,
+        PushDownPredicate,
         CollapseProject,
         RemoveRedundantProject) :: Nil
   }
 
   test("Column pruning after rewriting predicate subquery") {
-    val relation = LocalRelation('a.int, 'b.int)
-    val relInSubquery = LocalRelation('x.int, 'y.int, 'z.int)
+    withSQLConf(SQLConf.CONSTRAINT_PROPAGATION_ENABLED.key -> "false") {
```

Comment:
We need to modify this existing test?

Reply:
Yes, `spark.sql.constraintPropagation.enabled=false` to test `ColumnPruning`.
`spark.sql.constraintPropagation.enabled=true` to test `ColumnPruning`, `InferFiltersFromConstraints` and `PushDownPredicate`.

Reply:
Ah, I see. Thanks.